notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for value in family_values:\n",
+    "    print(value, ': ', round(train_data[train_data['family'] == value].groupby([\"date\"]).mean().skew()['sales'], 2))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sales_and_onpromotion_data = train_data.drop(['date', 'store_nbr', 'family'], axis=1)\n",
+    "sales_and_onpromotion_data.corr()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
+    "  plt.scatter(x='onpromotion', y='sales', data=current_value_data)\n",
+    "  plt.title(value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>It becomes understandable that promotions have a pretty good influence on sales INCREASING (in general, talking about all the data).<br>As we can see on the plots above, most of the 'family' values (but not all of them!) prove this fact. Interesting fact that 'books' didn't have any promotions during he whole period of observations.<br>Moreover, correlation is influenced by outliers, so this coefficient may not be accurate.<br>Nevertheless, 'onpromotion' feature is useful for the predictions.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "stores_data = pd.read_csv('C:\\\\Users\\\\Owner\\\\Desktop\\\\Proga\\\\REPOS\\\\mentorship_EPAM\\\\data\\\\stores.csv')\n",
+    "stores_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stores_data.info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>Store id doesn't help itself with the sales predictions, so it is necessary to replace the store id with the corresponding information about it</i>"
```

Comment:
I'm not sure about that: how many stores do we have in the data? Are we going to forecast for the same stores listed in the train set, or there might be some unseen stores in the test set? If the sets of stores in train / test are the same, we can potentially use it as a feature, since sales patterns may differ store-by-store.

Reply:
I agree, this feature can be useful for a predictive model. But I think, that using the store_nbr at the same time as all information about it (city, state, type, cluster) is an excess, because store ID uniquely defines all the other characteristics of the store. Maybe, I'm wrong..